src/components/Spinner/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Animated } from 'react-native';
+import Svg, { Circle } from 'react-native-svg';
+
+import { SpinnerSize, SpinnerVariant } from './types';
+import { SpinnerContainer, variants } from './styled';
+import useRotate from '../../hooks/useRotate';
+
+interface Props {
+    isVisible?: boolean;
+    size?: SpinnerSize;
+    variant?: SpinnerVariant;
+}
+
+/**
+ * Spinners should be shown when retrieving data or performing slow,
+ * help to reassure the user that the system is actively retrieving data.
+ */
+const Spinner: React.FC<Props> = ({
+    size = 'medium',
+    variant = 'brand',
+    isVisible,
+}) => {
+    const rotate = useRotate();
+
+    if (!isVisible) {
```

Comment:
in order to improve readability, we can do the opposite since negates are more difficult to understand:
```
if (isVisible) {
    // return component here
}
return null;
```